### ISSUE TITLE
Angular 2 Example Upgrade

### ISSUE THREAD

**Issue by bennett000** [rangle/rangle-starter#204]
Upgrade the Angular 2 example application to:

- [ ] conform to the angular.io style guide with respect to `NgModule`s
- [ ] use lazy loading for routes
- [ ] use AoT for production builds

**Comment by smithad15** [comment]
This still hasn't been completed in angular2-starter-example

**Comment by SethDavenport** [comment]
In many ways this overlaps with rangle/angular2-redux-example#247, which I've been working on: https://github.com/rangle/angular2-redux-example/pull/292

**Comment by SethDavenport** [comment]
The idea is that we'll get AoT and lazy loading if we move to the CLI, without having to maintain the toolchain. However one blocker to moving to the cli was the presence of the local auth server.  My PR replaces it with mock endpoints.

Once that's done, we can cli-ify it, and then move the folder structure around so it makes more sense/matches `ng lint` rules.

**Comment by SethDavenport** [comment]
I was having some issues with Robot but we should perhaps disable those tests and re work them eventually in terms of protractor?

**Comment by SethDavenport** [comment]
Especially since react-redux-example has the same robot test setup almost exactly if we want to keep a robot example around for QA

**Comment by smithad15** [comment]
Agreed that we should be maintaining a Protractor suite as a reference point. I can see that coming up more often in Angular-land than Robot
